Source/spinal-harmony.cs

```diff
@@ -1,314 +1,327 @@
-                    if (amp.Rotation.Opposite != instance.parent.Rotation) // Remember, engines vs amps are inverted
-                    {
-                        Log.Message("amps rotation in SpinalRecalc did not match parent rotation");
-                        amplifierCount = -1;
-                        break;
-                    }
-                    //This is... A way, to check I guess. (amp = ptp.gftwc<csm> returns thing, if that is *also* the center, 1-wide, must? be amp) 
-                    if (amp.Position == previousThingPos)
-                    {
-                        amplifierCount += 1;
-                        // Log.Message(string.Format("amp found, ampcount is {0}", AmplifierCount));
-                        ampBoost += ampComp.Props.ampAmount;
-                        ampComp.SetColor(spinalComp.Props.color);
-                    }
-                    //found emitter
-                    // Log.Message(string.Format("Looking for stackEnd results at {0}, result of: {1}", ampComp, ampComp.Props.stackEnd));
-                    else if (amp.Position == previousThingPos - vecMoveCheck && ampComp.Props.stackEnd)
-                    {
-                        // removed + 1 amp count because initializing to -1 feels kind of illegal
-                        foundNonAmp = true;
-                    }
-                    //found unaligned
-                    else
-                    {
-                        Log.Message(string.Format("amp.position: {0}, previouspos minus vec: {1} ", amp.Position, previousThingPos - vecMoveCheck));
-                        Log.Message("Found unaligned in SpinalRecalc");
-                        amplifierCount = -1;
-                        foundNonAmp = true;
-                    }
+                    //Log.Message($"amp found, toreturn.thrustAmp is {toReturn.thrustAmp}");
+                    toReturn.thrustAmp += ampComp.Props.thrustAmp;
+                    toReturn.fuelAmp += ampComp.Props.fuelUseAmp;
+                    toReturn.powerUseAmp += ampComp.Props.powerUseAmp;
+                    toReturn.supportWeight += ampComp.Props.supportWeight; //Currently none have a weight set but maybe I want one that does! 
+                    toReturn.fuelAllowAmp += ampComp.Props.fuelAllowAmp; // ok for sure none have this and... Will... They? Sounds illegal
+                    //ampComp.SetColor(spinalComp.Props.color);
                 }
-                Log.Message(string.Format("Found amps on spinal engine: {0}, boost: {1}", amplifierCount, ampBoost));
+                //found emitter
+                else if (buildingPointer.Position == previousThingPos - vecMoveCheck && ampComp.Props.stackEnd)
+                {
+                    foundNonAmp = true;
+                    foundFuelStart = true;
+                    toReturn.supportWeight += ampComp.Props.supportWeight; // likely first *real* supportWeight
```

Comment:
Hrm I don't think? This is needed because the next check is an else anyhow so it would never be evaluated due to the else if happening? Would it be better to explicit-beak in the if (buildingPointer.Position == previousThingPos), and the elseif, for some reason?